### DISCUSSION TITLE
Space vs orientation

### DISCUSSION THREAD

**Discussion by kretes** [Project-MONAI/MONAI#6808] [Q&A]
Hello,

I have Nifti data converted from dicom using dicom2niftii with default settings. This means that the data is converted to LAS orientation as per
https://github.com/icometrix/dicom2nifti/blob/dfc18475eb506885a065fe212e8c18791fbc8b04/dicom2nifti/convert_dicom.py#L64

Now, when I read this scan using MONAI - what I get is that the space gets set to 'RAS' 
(either by NibabelReader https://github.com/Project-MONAI/MONAI/blob/c4ff70b3df52f16e4a2155217f8e5f3aa6be8ce6/monai/data/image_reader.py#L936 or when the default `affine_lps_to_ras=True` in ITKReader)
At the same time my affine matrix is sth like:
```
tensor([[  -0.7148,    0.0000,    0.0000,  197.2000],
         [  -0.0000,    0.7148,    0.0000, -177.3853],
         [   0.0000,   -0.0000,    1.2500, -278.0000],
         [   0.0000,    0.0000,    0.0000,    1.0000]], dtype=torch.float64),
```
so the orientation of LAS. while `img.meta['space']` says 'RAS'.

I think this is all intended, as I dig up an old comment https://github.com/Project-MONAI/MONAI/pull/4790#issuecomment-1198893285 by @wyli saying that `img.meta['space']` isn't `orientation`.

I wonder if it is safe to say that `space` is a `reader_space` (as mentioned in the PR) as it can't be altered for a loaded image, right?
 
For a clearer picture - should there be an `orientation` field in the image, which gets calculated from the `affine`, like mentioned in https://github.com/Project-MONAI/MONAI/discussions/4224 ?

Wondering about places in MONAI's codebase when there is a check of two images being co-registered. I understand that `space` isn't important there, as the `affine` does define the img2world transformation and the images need to have the same `affine, Right?

**Comment by KumoLiu** [comment]
Hi @kretes, as Wenqi said, the `img.meta['space']` here is the base coordinate type. Orientation is relative to the base coordinate type and computed from the affine.
And for this arg `affine_lps_to_ras `, it mainly adds a flag to switch the different default base coordinate type between different readers. https://github.com/Project-MONAI/MONAI/issues/4789
Hope it helps, thanks!

**Reply by kretes** [reply]
Since this is something that brings confusion I need to iterate on it:
If the `img.meta['space']` is the base coordinate type and `affine` should be interpreted w.r.t. this space - then I don't get the usefulness of `affine_lps_to_ras`.

when `affine_lps_to_ras=True` - what happens is that `img.meta['space']` is set to RAS and `affine` gets flipped between lps and ras. (e.g. ITKReader https://github.com/Project-MONAI/MONAI/blob/4c22a27aa5bc0b2d9c41f6d9a3201cc8d99a53ed/monai/data/image_reader.py#L331-L332 )
But now - since both `space` and `affine` changed - we would get exactly the same meaning of the data as with `affine_lps_to_ras=False`. Flipping base space and `affine` has no effect of the final orientatation if it would be calculated from both of them. Am I missing something?

How I look at it - depending on the data we read - we need to stay with `affine_lps_to_ras=True` or in some rare cases set `affine_lps_to_ras=False`. Whatever we do - the `affine` we get in the end will always be w.r.t. the RAS space. Will it be 'correct' depends on the user setting `affine_lps_to_ras`. `img.meta['space']` will just remain a record on how was the data read.

**Reply by KumoLiu** [reply]
Hi @kretes, 
- Affine is calculated by using direction(based on base coordinate type) and spacing.
https://github.com/Project-MONAI/MONAI/blob/4c22a27aa5bc0b2d9c41f6d9a3201cc8d99a53ed/monai/data/image_reader.py#L329-L330
- And since you set `affine_lps_to_ras=True` in ITKReader, and ITKReader by default use LPS https://dicom.innolitics.com/ciods/mr-image/image-plane/00200037, so we flip the first two spatial dimensions in L332.
https://github.com/Project-MONAI/MONAI/blob/4c22a27aa5bc0b2d9c41f6d9a3201cc8d99a53ed/monai/data/image_reader.py#L322
https://github.com/Project-MONAI/MONAI/blob/4c22a27aa5bc0b2d9c41f6d9a3201cc8d99a53ed/monai/data/image_reader.py#L331-L332
- Whether to set `affine_lps_to_ras` you can refer to the docstring.

https://github.com/Project-MONAI/MONAI/blob/4c22a27aa5bc0b2d9c41f6d9a3201cc8d99a53ed/monai/data/image_reader.py#L175-L176

- `space` here only helps record which `base coordinate type` to use.

Hope it helps, thanks!
